sdk/core/azure-core/inc/azure/core/exception.hpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Define RequestFailedException. It is used by HTTP exceptions.
+ */
+
+#pragma once
+
+#include <stdexcept>
+
+namespace Azure { namespace Core {
+  struct RequestFailedException : public std::runtime_error
```

Comment:
Should we add status and error code members, similar to https://github.com/Azure/azure-sdk-for-net/blob/cd97a7ffcb948d5a6e21f3c3fc6ef0d4737b0424/sdk/core/Azure.Core/src/RequestFailedException.cs#L17-L25

Reply:
I don't think so. `RequestFailedException` is a base class of all kinds of exception. there's might not be a status or error code yet, for example, fail when resolving host.

Reply:
How come .NET has them? If there is no status or error code, can we map that to a particular default value?

I suppose these can be added later, as needed.